### ISSUE TITLE
What is the purpose of --cache-file?

### ISSUE THREAD

**Issue by Strat-** [mpv-player/mpv#2083]
I'm trying to stream a large video file, and I want it to cache the entire video to file, rather store it in RAM. This is my configuration:

```
cache-file=/home/user/mpv-cache
cache-file-size=20971520
```

...and I see the file has been created, but mpv (seemingly) still has a small cache, maxing out at ~75MB at a time.

If I set `cache` to a large size, it caches more of the video, but it caches to RAM.

What does `cache-file` actually do? Am I doing something wrong here?

**Comment by wm4** [comment]
The cache file is an additional cache between the RAM cache and the network stream. How much of the cache file uses is not displayed anywhere, and it probably works correctly. Note that the cache file is reset every time a stream is opened.

**Comment by Strat-** [comment]
...but when the RAM cache fills up, MPV stops caching the file. I can tell my seeing my network IO graph. It doesn't seem that mpv is caching to the file at all.

**Comment by wm4** [comment]
Yeah. You actually need to play the file to advance the RAM cache position.

**Comment by Strat-** [comment]
You seem to be misunderstanding. What I'm looking for is for mpv to cache the entire stream, something like this:


`[=====++++++++++++++++----]`

* `=`: RAM cache
* `+`: File cache
* `-`: Uncached

I've been under the impression that `cache-file` is what does that, or am I wrong?

**Comment by Strat-** [comment]
tl;dr: When the RAM cache is full, isn't the cache file supposed to start caching whatever isn't the RAM cache?

**Comment by wm4** [comment]
The file cache isn't "active" - it just caches what the layers above it (RAM cache or demuxer) request from it. I don't think the behavior you want is currently possible. Yes, maybe it could be implemented somehow.

**Comment by Strat-** [comment]
So the file cache currently stores only what has already been cached in RAM/demuxer, correct?

**Comment by wm4** [comment]
Correct.

**Comment by Strat-** [comment]
Alright, thanks for clearing that up.
